core/trino-main/src/main/java/io/trino/sql/analyzer/Analyzer.java

```diff
@@ -80,18 +80,23 @@ public Analyzer(
 
     public Analysis analyze(Statement statement)
     {
-        return analyze(statement, false);
+        return analyze(statement, false, false);
     }
 
     public Analysis analyzeForDescribe(Statement statement)
     {
-        return analyze(statement, true);
+        return analyze(statement, true, false);
     }
 
-    private Analysis analyze(Statement statement, boolean isDescribe)
+    public Analysis analyzeForExplain(Statement statement)
+    {
+        return analyze(statement, false, true);
+    }
+
+    private Analysis analyze(Statement statement, boolean isDescribe, boolean isExplain)
```

Comment:
`isDescribe` and `isExplain` should be part of an enum. It's invalid to have both set and it's not clear from the code what would/should happen if they are.